### ISSUE TITLE
Excel - Json - Joomla

### ISSUE THREAD

**Issue by Coral-66** [NUKnightLab/TimelineJS3#793]
A contribution to TimelineJS3

3 years ago I developed an Excel Based (VBA driven) routine to convert an Excel Worksheet to a JSON file which can be used instead of GoogleSheets; The results can be viewed at : https://oathlawtannadice.org.uk/index.php/history/time-line-stories.html#event-things-are-about-to-change- (TimelineJS3 Version 3.6.5) (The site is develop using Joomla, TimelineJS3 called from Joomla Article)

Currently looking to move the site to Joomla V 4 and checking out TimelineJS3. 
Installed Locally (WAMPSERVER/Locahost) the Timelines display using latest version of TimeLineJ3 (3.8.21) with minor issues as noted:

a. TimelineJS3 options parameter  'Ease' causes error - Commenting out resolves
b. Re-write of Javascript prevents coloured markers - may be resolved in future
c. Re-write of Javascript uses SANATIZE for html - needed slight change to HTML on one slide - move 'Style' to CSS

In Summary the EXCEL to JSON routine is valid for lastest version of TimelineJS3
The EXCEL routine also has a function to create a Slide Entry in a vertical format and insert into the standard format
Anyone interested in the above can request the EXCEL Sheets / Joomla details

Personal Note: Algol 60 at School / Various at Uni / Coral66 first employment
Not too well conversant with Javascript, Node.JS etc.
Not a member of Tannadice Church - Project was favour for friend
 
Attached are JSON file for TimeLine : Tannadice Church (1500 - 1900 AD) (rename OTProject01.txt to OTProject01.json
Also Some Screen Shots

[OT-Project01.txt](https://github.com/NUKnightLab/TimelineJS3/files/9478094/OT-Project01.txt)
**Sample Web Page**
![Page from Timeline 1838 -1841 ](https://user-images.githubusercontent.com/112514409/188153812-7c135604-ea5d-4b36-8763-e44c9afa7f6a.jpg)
**Various Screen Shots of Excel**
![Excel Sheet (left)](https://user-images.githubusercontent.com/112514409/188155590-bb1e7e46-e969-4405-8ecf-3f27b21333ed.jpg)
![Excel Sheet (Middle)](https://user-images.githubusercontent.com/112514409/188155635-1df7cfbf-bd3b-4c2b-9818-3bc5d5b8821f.jpg)
![Excel Sheet (Right-75%)](https://user-images.githubusercontent.com/112514409/188155668-23fb9e74-9ebf-42e2-9d88-3fed893203a0.jpg)
![Excel Slide Create Vertical](https://user-images.githubusercontent.com/112514409/188155710-2dfc3e8b-39ce-442e-9ef7-3844681b641d.jpg)

**Comment by JoeGermuska** [comment]
Interesting. It's not clear to me, however, the form which this contribution would take. Would this be like a template Excel file we could offer from the `/contrib/` directory? Something else?

**Comment by Coral-66** [comment]
The full EXCEL worksheet can be provided which includes the VBA to generate the JSON file. One of the worksheets has a Parameter which gives the local folder etc. where you want the JSON file create. For LOCALHOST this allows immeadiate running of the TIMELINE. For WEB based the file would need transferring to the site.  This can be done in VBA (I think) or autometed FTP.
At present the EXCEL VBA just takes what is entered but could be enhanced to do some basic syntax checking/error checking of the line enteries  (Eg. Added some code to eliminate situation where Start Date/End Date were the same - was throwing debug error in 3.8.21).
Inital development was to remove need for GoogleSheets, Ease JSON creation and allow Local Development & Testing

**Comment by JoeGermuska** [comment]
If you would be willing to also write a concise README markdown document guiding people on how to use the Excel worksheet, we could include both in a subdirectory of `/contrib/`. If you could provide them as a pull request, that would be great, but if that's a headache for you, you could also attach the files to a comment here, or we could figure something out.

If you do a pull request, let's put the assets in `/contrib/excel-to-json/`; you should be able to include screenshots if you think they are necessary for the README.

**Comment by Coral-66** [comment]
Happy to prepare a README document, being concise may be issue (should manage as most of detail already on KnightLab site).
Will read info on Github / Pull request and attempt, if issue will add here. 
(Should complete in 24/48 hours) 
Note: Worksheet developed using Excel 2010 so should work on later versions which support VBA macros

**Comment by JoeGermuska** [comment]
of course, use links where they make sense, such as in referring to handling the JSON file -- that part will, honestly, be outside the comfort zone for most TimelineJS users, but then, most TimelineJS users aren't looking in the `/contrib/` directory of the github repo either...

and no rush from our perspective. We'd rather have it slow and correct than rushed and in need of edits and follow-up!

**Comment by Coral-66** [comment]
My account with Github was 'Flagged' (Apparently a regular occurrence with new accounts). It has now been cleared and as such content here is now visible again.
Anticipate creating 'Push' request with content from above comment by tomorrow.(Monday 12th Sept. 2022)

**Comment by Coral-66** [comment]
Logic to create a 'Pull Request' proving challenging. README info & files in comment above. The text and files can also be found on my Github account'
Note: This is a standalone utility designed to ease the creation of JSON files for TimelineJS3. No technical knowledge of MS Excel is required.
(Joe Germuska : Happy to accept instruction/guide to create 'Pull Request'
